### DISCUSSION TITLE
Integration to a TypeScript project

### DISCUSSION THREAD

**Discussion by bjost2s** [matteobruni/tsparticles#1293] [General]
I would like to integrate tspraticles into my TypeScript project, but types are missing. Any way to get the types via npm?

**Comment by matteobruni** [comment]
All the project is made using TypeScript, which types are missing? They are all compiled and available in the npm package

**Reply by bjost2s** [reply]
Thank you for your quick response. I have installed it with 
`npm install tsparticles`
than`import { tsParticles } from "tsparticles";`
here is the output from tsc:
 ERROR in /home/bjost/Git/RobCoWeb/src/Program.ts(14,29)
      TS2307: Cannot find module 'tsparticles' or its corresponding type declarations.

**Reply by matteobruni** [reply]
I did a sample using CodeSandobx some time ago, you can check out it here: https://codesandbox.io/s/tsparticles-c6kim?file=/src/index.ts

It was made using TypeScript and the import works fine as expected.

If it's possible try attaching a zip with the issue

**Reply by bjost2s** [reply]
Thank you so much! It works now. I have changed the  "module" property in the tsconfig.json from "es6" to  "commonjs" like in yours and everything is working as expected. 
I have started a project with matter.js but would like to check, if tsparticle fits better. Now I can start working on this :-)

**Reply by bjost2s** [reply]
After doing some experiments I with your setup, there is a new problem:
`Uncaught (in promise) TypeError: Cannot read property 'numberOfItems' of undefined`
`at parsePaths (PolygonMaskInstance.js:49)`
    `at PolygonMaskInstance.parseSvgPath (PolygonMaskInstance.js:274)`
    `at PolygonMaskInstance.<anonymous> (PolygonMaskInstance.js:288)`
   ` at Generator.next (<anonymous>)`
    `at fulfilled (PolygonMaskInstance.js:5)`
This error message appears at runtime in the console. I only have changed the tsparticle config from the simple one to the one from [here](https://codepen.io/matteobruni/embed/MWaePMa?height=600&default-tab=result&embed-version=2)

**Reply by matteobruni** [reply]
The polygonMask feature requires `pathseg` library

You can read more here: https://github.com/matteobruni/tsparticles#dependencies

Or here: https://github.com/matteobruni/tsparticles/issues?q=is%3Aissue+pathseg
